homeassistant/components/frontier_silicon/config_flow.py

```diff
@@ -0,0 +1,237 @@
+"""Config flow for Frontier Silicon Media Player integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from afsapi import AFSAPI, ConnectionError as FSConnectionError, InvalidPinException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import ssdp
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import (
+    CONF_PIN,
+    CONF_WEBFSAPI_URL,
+    DEFAULT_PIN,
+    DEFAULT_PORT,
+    DOMAIN,
+    SSDP_ATTR_SPEAKER_NAME,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+    }
+)
+
+STEP_DEVICE_CONFIG_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(
+            CONF_PIN,
+            default=DEFAULT_PIN,
+        ): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Frontier Silicon Media Player."""
```

Comment:
Would it be possible to reuse code a bit more in this class? The logic for user, ssdp and import looks pretty the same. Have a look at some other config flows to see how they call `async_step_user` from ssdp for example.